app/types.go

```diff
@@ -44,22 +44,20 @@ type SignalingServer interface {
 	ServeDeviceFiles(zone string, host string, params DeviceFilesRequest, user UserInfo) error
 }
 
-type DeviceDesc struct {
-	// The (internal) network address of the host where the cuttlefish device is
-	// running. The address can either be an IPv4, IPv6 or a domain name.
-	Addr string
-	// The id under which the cuttlefish device is registered with the host
-	// orchestrator (can be different from the id used in the cloud orchestrator)
-	LocalId string
-}
-
 type InstanceManager interface {
-	DeviceFromId(zone string, host string, name string, user UserInfo) (DeviceDesc, error)
+	// Returns the (internal) network address of the host where the cuttlefish device is
+	// running. The address can either be an IPv4, IPv6 or a domain name.
+	GetHostAddr(zone string, host string, user UserInfo) (string, error)
+	// Creates a host instance.
 	CreateHost(zone string, req *apiv1.CreateHostRequest, user UserInfo) (*apiv1.Operation, error)
 	// Closes the connection with the underlying API
 	Close() error
 }
 
+var ErrBadCreateHostRequest = NewBadRequestError("invalid CreateHostRequest", nil)
+var ErrHostInstanceNotOwnedByUser = NewForbiddenError("host instance not owned by user", nil)
+var ErrMissingNetworkInterface = NewInternalError("host instance missing a network interface", nil)
```

Comment:
what stops code from another package from doing
```
gcp.ErrBadCreateHostRequest = <something else>
```
??

IIUC the only reason you need these variables is so you can test with `errors.Is()`, but the same effect can be achieved by making these types instead of variables and checking with `errors.As()`.